plugin/trino-hive/src/main/java/io/trino/plugin/hive/parquet/ParquetFileWriterFactory.java

```diff
@@ -127,6 +145,44 @@ public Optional<FileWriter> createFileWriter(
                     HIVE_PARQUET_USE_LEGACY_DECIMAL_ENCODING,
                     HIVE_PARQUET_USE_INT96_TIMESTAMP_ENCODING);
 
+            Optional<Function<ParquetWriteValidation, ConnectorPageSource>> validationInputFactory = Optional.empty();
+            if (isParquetOptimizedWriterValidate(session)) {
+                validationInputFactory = Optional.of(writeValidation -> {
+                    try {
+                        long fileSize = fileSystem.getFileStatus(path).getLen();
+                        List<HiveColumnHandle> columns = IntStream.range(0, fileColumnNames.size()).boxed()
```

Comment:
Does it handle nested type?

Reply:
It does work for nested types, that's verified by `TestHiveFileFormats#testOptimizedParquetWriter`.
Was there anything that you expected to be different about this code in case of nested types ?